### PR TITLE
Mecha Ramming!

### PR DESCRIPTION
**Pull request by KazeEspada**
Certain Mecha now will automatically attack whatever they hit. At the moment, this is only Gygaxes in run mode, and anything in the Maurader family(Mauler, Maurader, and Seraph).

Gygax's no longer damage themselves while running, but now severely drain their cells to maintain their impressive speed.

Fixed a text issue with mecha wall smash.

### CODE REVIEW

File: code/game/mecha/mecha.dm [hunk 1]
```diff
@@ -545,6 +540,37 @@
 		if(yes)
 			if(..()) //mech was thrown
 				return
+			if(bumpsmash)
+				var/mecha = src
+				obstacle.mech_melee_attack(mecha)
+				if(!obstacle || (obstacle && !obstacle.density))
+					step(src,dir)
+				/*var/success = 0
```
Comment (optimumtact) [inline code/game/mecha/mecha.dm]: You thought we wouldn't see, but we did

File: code/game/turfs/simulated/walls.dm [hunk 2]
```diff
@@ -88,9 +88,9 @@
 	if(M.damtype == "brute")
 		playsound(src, 'sound/weapons/punch4.ogg', 50, 1)
 		visible_message("<span class='danger'>[M.name] has hit [src]!</span>")
-		if(prob(hardness) && M.force > 20)
+		if(prob(hardness + M.force) && M.force > 20)
 			dismantle_wall(1)
-			visible_message("<span class='warning'>[src.name] smashes through the wall!</span>")
+			visible_message("<span class='warning'>[M.name] smashes through the wall!</span>")
```
Comment (optimumtact) [inline code/game/turfs/simulated/walls.dm]: OOOOOOH YEAHHHHHHHH

SMASH

**Comment by phil235** [comment]
Did you test if smashing into things has a proper attack cooldown?

**Comment by AdamElTablawy** [comment]
>run aimlessly around in a mech

>slap everyone in your way out of the way on accident

>get lynched

**Comment by GunHog** [comment]
I cannot wait to see WarDec Ops ROFLSMASH through their drop pod and the station!

**Comment by KazeEspada** [comment]
It uses the mech step cooldown which is slower for most mechs, and its not supposed to use the attack cooldown.

**Comment by xxalpha** [comment]
When will mechas have AT fields.

**Comment by Zilenan91** [comment]
Oh man it drains the battery that's already infinite because of reactors anyways, what a downside

**Comment by KazeEspada** [comment]
It drains much faster than any generators

**Comment by WJohn** [comment]
God this makes the gygax even more of "best mech" than other choices.

**Comment by Zilenan91** [comment]
The Durand is still better. It WILL shit on everyone even through this.

**Comment by KazeEspada** [comment]
Durand got no changes.

**Comment by optimumtact** [comment]
This should do damage to the mech to balance it a better

**Comment by KazeEspada** [comment]
No, it already drains the cell severely.

**Pull request merged**